src/plugins/discover/public/application/components/sidebar/discover_field_details.tsx

```diff
@@ -39,6 +43,36 @@ export function DiscoverFieldDetails({
   onAddFilter,
 }: DiscoverFieldDetailsProps) {
   const warnings = getWarnings(field);
+  const [showVisualizeLink, setShowVisualizeLink] = useState<boolean>(false);
+  const [visualizeLink, setVisualizeLink] = useState<string>('');
+
+  useEffect(() => {
+    isFieldVisualizable(field, indexPattern.id, details.columns).then(
+      (flag) => {
+        setShowVisualizeLink(flag);
+        // get href only if Visualize button is enabled
+        getVisualizeHref(field, indexPattern.id, details.columns).then(
+          (uri) => {
+            if (uri) setVisualizeLink(uri);
+          },
+          () => {
+            setVisualizeLink('');
+          }
+        );
+      },
+      () => {
+        setShowVisualizeLink(false);
+      }
+    );
+  }, [field, indexPattern.id, details.columns]);
+
+  const handleVisualizeLinkClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    if (!event.defaultPrevented) {
+      // regular link click. let the uiActions code handle the navigation and show popup if needed
+      event.preventDefault();
+      triggerVisualizeActions(field, indexPattern.id, details.columns);
```

Comment:
@stratoula thx for the changes 🙏  I think it makes sense to finish the review once #73226 is merged, then also all legacy css classes should be gone 🥳 . BTW what kind of popover could be displayed?

Reply:
It's the popover that uiactions have implemented. It appears when we have more than one actions to the same trigger. Same as drill downs

Reply:
thx for the context 👍 , so now that #73226 was merged , this is unblocked for fixing conflicts + final review

Reply:
Doing it right now 🥊

Reply:
@kertal, conflicts fixed :)